components/chat_widget/src/components/ocs-chat/ocs-chat.tsx

```diff
@@ -705,6 +725,32 @@ export class OcsChat {
     }
   }
 
+  private getOrGenerateUserId(): string {
+    if (this.userId) {
+      return this.userId;
+    }
+
+    if (this.generatedUserId) {
+      return this.generatedUserId;
+    }
+
+    const storageKey = `ocs-user-id-${this.chatbotId}`;
+    const stored = localStorage.getItem(storageKey);
+    if (stored) {
+      this.generatedUserId = stored;
+      return stored;
+    }
+
+    const array = new Uint8Array(9);
+    window.crypto.getRandomValues(array);
+    const randomString = Array.from(array, byte => byte.toString(36)).join('').substr(0, 9);
+    const newUserId = `user_${Date.now()}_${randomString}`;
```

Comment:
nit: update prefix to be consistent with existing scheme (https://github.com/dimagi/open-chat-studio/blob/83c42ac3871131d598e8cb50aa24dd11a7d3641a/apps/experiments/models.py#L1325)

```suggestion
    const newUserId = `ocs:${Date.now()}_${randomString}`;
```